src/typeahead/typeahead.ts

```diff
@@ -88,6 +124,9 @@ export class NgbTypeahead implements ControlValueAccessor,
   private _subscription: Subscription;
   private _userInput: string;
   private _valueChanges: Observable<string>;
+  private _input: NgbTypeaheadInitParams['input$'];
```

Comment:
Didn't know it worked!
But still would prefer to see the type with my own eyes: `private _input: Observable<...>`
Maybe it's just me though

Reply:
On one hand I agree with you because indeed in some contexts it's not even easy to go check what `NgbTypeaheadInitParams['input$']` actually means (like here where GitHub displays only a few lines).

On the other hand, the semantics and maintenance complexity would change if I don't use a reference anymore. It wouldn't be linked anymore. It's not a very big deal though in this case.

PS: I would have expected to have a clue from the tools like in Visual Studio Code, for instance something like `NgbTypeaheadInitParams['input$'] => Observable<HTMLInputElementEvent>` in a tooltip over the type reference, but it's not even the case.